File: src/etch/bindings/java/msg/Validator_object.java
/* $Id$
 *
 * Created by wert on Aug 27, 2007.
 *
 * Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
 */

package etch.bindings.java.msg;

import etch.bindings.java.transport.fmt.TypeCode;

/**
 * Validator for object.
 */
public class Validator_object extends Validator
{
	@Override
	public Byte checkValue( Object value )
	{
		return TypeCode.ANY;
	}

	@Override
	public Validator elementValidator()
	{
		return this;
	}

	@Override
	public boolean validate( Object value )
	{
		return true;
	}

	public static Validator get( int nDims )
	{
		if (inst == null)
			inst = new Validator_object();
		return inst;
	}
	
	private static Validator_object inst;
}

File: src/etch/bindings/csharp/msg/Validator_object.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Msg
{
    public class Validator_object : TypeValidator
    {

        private static Validator_object[] validators = new Validator_object[MAX_CACHED];

        public override Validator ElementValidator()
        {
            return nDims > 0 ? Get(nDims - 1) : this;
        }

        private Validator_object(int nDims)
            : base(typeof (object), typeof (object), nDims, string.Format("object[{0}]", nDims), true)
        {
            
        }

        public static Validator_object Get( int nDims )
        {
            CheckDims(nDims);

            if (nDims >= validators.Length)
                return new Validator_object(nDims);

            Validator_object v = validators[nDims];

            if ( v == null )
                v=validators[nDims]= new Validator_object(nDims);

            return v;
        }      
    }
}
